src/models/hooks/objective.js

```diff
@@ -0,0 +1,195 @@
+// import { Op } from 'sequelize';
+import { auditLogger } from '../../logger';
+
+const autoPopulateObjectiveTemplateId = async (sequelize, instance, options) => {
+  // eslint-disable-next-line no-prototype-builtins
+  if (!instance.hasOwnProperty('objectiveTemplateId')
+  || instance.objectiveTemplateId === null
+  || instance.objectiveTemplateId === undefined) {
+    let regionId = null;
+    // eslint-disable-next-line no-prototype-builtins
+    if (instance.hasOwnProperty('goalId')
+    && instance.goalId !== null
+    && instance.goalId !== undefined) {
+      const goal = await sequelize.models.Goal.findOne({
+        where: { id: instance.goalId },
+        include: [
+          {
+            model: sequelize.models.Grant,
+            as: 'grant',
+            attributes: ['regionId'],
+          },
+        ],
+      });
+      regionId = goal.grant.regionId;
+      // eslint-disable-next-line no-prototype-builtins
+    } else if (instance.hasOwnProperty('otherEntityId')
+    && instance.otherEntityId !== null
+    && instance.otherEntityId !== undefined) {
+      try {
+        const otherEntity = await sequelize.models.OtherEntity.findAll({
+          attributes: [
+            [sequelize.literal('array_agg(`ActivityReports`.regionId)'), 'regionIds'],
+          ],
+          where: { id: instance.otherEntityId },
+          include: [
+            {
+              model: sequelize.models.ActivityReport,
+              as: 'activityReports',
+              attribites: [],
+            },
+          ],
+          group: ['id'],
+        });
+        if (otherEntity.regionIds.length === 1) {
+          [regionId] = otherEntity.regionIds;
+        }
+      } catch (err) {
+        auditLogger.error(JSON.stringify(err));
+        throw err;
+      }
+    }
+
+    const objectiveTemplate = await sequelize.models.ObjectiveTemplate.findOrCreate({
+      where: { hash: sequelize.fn('md5', sequelize.fn('NULLIF', sequelize.fn('TRIM', instance.title), '')), regionId },
```

Comment:
So you want to gen the md5 in js instead of having the db use the standard md5 function built into the db? This is a distinct way to reproducibly generate md5s.

Reply:
I'm saying we move the text that gets sent to the DB to generate md5s into it's own function. That way what we pass to the DB is defined in one place